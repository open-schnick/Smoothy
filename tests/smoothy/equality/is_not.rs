use crate::failing_assertion;
use smoothy::{assert_that, EqualityAssertion};

mod succeeds {
    use super::*;

    #[test]
    fn with_numbers() {
        assert_that(21u8).is_not(42);
    }

    #[test]
    fn with_strings() {
        assert_that(String::from("Hello World!")).is_not(String::from("Hello There!"));
    }

    #[test]
    fn with_string_slices() {
        assert_that("Hello World!").is_not("Hello There!");
    }
}

mod fails {
    use super::*;

    #[test]
    fn with_matching_values() {
        failing_assertion!({
            assert_that(21u8).is_not(21);
        });
    }
}
