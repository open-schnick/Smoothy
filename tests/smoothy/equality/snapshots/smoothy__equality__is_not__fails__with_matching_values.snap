---
source: tests/smoothy/equality/is_not.rs
expression: assertion_failed_output
---
[31mAssertion failed![0m

Expected
  [31m21[0m
to not match
  [32m21[0m
